src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -103,24 +105,36 @@ internal sealed class RSession : IRSession, IRCallbacks {
 
         public Task<IRSessionInteraction> BeginInteractionAsync(bool isVisible = true, CancellationToken cancellationToken = default(CancellationToken)) {
             if (!_isHostRunning) {
-                return CanceledInteractionTask;
+                return CanceledBeginInteractionTask;
             }
 
             RSessionRequestSource requestSource = new RSessionRequestSource(isVisible, _contexts, cancellationToken);
```

Comment:
That can be a problem. I think we haven't run into this because we don't have cases when more than one request is scheduled.